src/adapter/redstone/RedstoneCoreOracle.sol

```diff
@@ -28,91 +43,98 @@ contract RedstoneCoreOracle is PrimaryProdDataServiceConsumerBase, BaseAdapter {
     /// @param _feedId The identifier of the price feed.
     /// @param _feedDecimals The decimals of the price feed.
-    /// @param _maxPriceStaleness The maximum allowed age of the Redstone price in `updatePrice`.
-    /// @param _maxCacheStaleness The maximum allowed age of the cached price in `_getQuote`.
-    /// @dev Since Redstone prices are verified locally, callers can pass data up to `maxPriceStaleness` seconds old.
-    /// It effectively imposes a deadline on the transaction, so a staleness window that is too short
-    /// increases the probability that the transaction reverts, especially during chain congestion.
-    constructor(
-        address _base,
-        address _quote,
-        bytes32 _feedId,
-        uint8 _feedDecimals,
-        uint256 _maxPriceStaleness,
-        uint256 _maxCacheStaleness
-    ) {
-        if (_maxPriceStaleness > MAX_PRICE_STALENESS_UPPER_BOUND) {
-            revert Errors.PriceOracle_InvalidConfiguration();
-        }
-        if (_maxCacheStaleness > MAX_CACHE_STALENESS_UPPER_BOUND) {
-            revert Errors.PriceOracle_InvalidConfiguration();
-        }
-        if (_maxCacheStaleness > _maxPriceStaleness) {
-            revert Errors.PriceOracle_InvalidConfiguration();
-        }
+    /// @param _maxStaleness The maximum allowed age of the Redstone price in `updatePrice`.
+    /// @dev Since Redstone prices are verified locally, callers can pass data up to `maxStaleness` seconds old.
+    /// If `maxStaleness` is too short, the update transaction may revert.
+    constructor(address _base, address _quote, bytes32 _feedId, uint8 _feedDecimals, uint256 _maxStaleness) {
+        if (_maxStaleness > MAX_STALENESS_UPPER_BOUND) revert Errors.PriceOracle_InvalidConfiguration();
 
         base = _base;
         quote = _quote;
         feedId = _feedId;
         feedDecimals = _feedDecimals;
-        maxPriceStaleness = _maxPriceStaleness;
-        maxCacheStaleness = _maxCacheStaleness;
+        maxStaleness = _maxStaleness;
         uint8 baseDecimals = _getDecimals(base);
         uint8 quoteDecimals = _getDecimals(quote);
         scale = ScaleUtils.calcScale(baseDecimals, quoteDecimals, _feedDecimals);
+        cache = Cache({price: 0, priceTimestamp: 0, updatePriceContext: FLAG_UPDATE_PRICE_EXITED});
     }
 
     /// @notice Ingest a signed update message and cache it on the contract.
-    /// @dev Validation logic inherited from PrimaryProdDataServiceConsumerBase.
+    /// @dev Validation logic inherited from `PrimaryProdDataServiceConsumerBase`.
+    /// During execution the context flag is set to `FLAG_UPDATE_PRICE_ENTERED`.
+    /// The execution context is checked in `validateTimestamp` to revert external calls.
     function updatePrice() external {
-        // Use the cache if it has not expired.
-        if (block.timestamp <= maxCacheStaleness + cacheUpdatedAt) return;
+        cache.updatePriceContext = FLAG_UPDATE_PRICE_ENTERED;
```

Comment:
Maybe `updatePrice` should verify that the price context is `FLAG_UPDATE_PRICE_EXITED`, in case `updatePrice` re-enters itself somehow.

Reply:
It cannot reenter however as `getOracleNumericValueFromTxMsg` is view

Reply:
Got it. But, if the code and/or redstone library is ever changed, this would become an ineffective re-entrancy guard. Since it costs almost nothing, it seems like it would be a good idea to follow the usual pattern of making sure that a lock is not already acquired prior to acquiring it.